### PR TITLE
Moved url, username, and password on ExternalIntegration to settings, and fixed a CSRF token problem

### PR DESCRIPTION
**Pull request by aslagle**
This simplifies the admin interface code using https://github.com/NYPL-Simplified/server_core/pull/528.

I also fixed an admin interface bug I noticed due to the CSRF token changing on every page load. When you opened a second tab, and then went back to the first tab and tried to submit a form, you'd get a CSRF token error since that tab's html still had the old token, but your cookie had the new token.

### CODE REVIEW

File: api/admin/controller.py [hunk 1]
```diff
@@ -1008,38 +1008,36 @@ def collections(self):
...
 
         protocols.append({
             "name": ExternalIntegration.ONE_CLICK,
             "fields": [
-                { "key": "password", "label": _("Basic Token") },
+                { "key": ExternalIntegration.PASSWORD, "label": _("Basic Token") },
                 { "key": "external_account_id", "label": _("Library ID") },
-                { "key": "url", "label": _("URL") },
-                { "key": "ebook_loan_length", "label": _("eBook Loan Length") },
-                { "key": "eaudio_loan_length", "label": _("eAudio Loan Length") },
```
Comment (aslagle) [inline api/admin/controller.py]: I believe these two settings were removed from oneclick in an earlier branch.

File: api/admin/routes.py [hunk 2]
```diff
@@ -379,7 +379,7 @@ def admin_view(collection=None, book=None, **kwargs):
     else:
         home_url = None
 
-    csrf_token = app.manager.admin_sign_in_controller.generate_csrf_token()
+    csrf_token = flask.request.cookies.get("csrf_token") or app.manager.admin_sign_in_controller.generate_csrf_token()
```
Comment (aslagle) [inline api/admin/routes.py]: If you already have a CSRF token cookie, you keep it, but if not you get a new one.

File: api/admin/controller.py (hunk 1 above)
Comment (leonardr) [inline api/admin/controller.py]: That's correct, but every Collection (regardless of its protocol) can now have a default_loan_period and a default_reservation period.

**Comment by leonardr** [comment]
:+1:

**Pull request merged**